app/assets/javascripts/reducers/uploads.js

```diff
@@ -1,11 +1,14 @@
-import { RECEIVE_UPLOADS, SORT_UPLOADS, SET_VIEW, GALLERY_VIEW, FILTER_UPLOADS } from '../constants';
+import _ from 'lodash';
+import { RECEIVE_UPLOADS, SORT_UPLOADS, SET_VIEW, GALLERY_VIEW, FILTER_UPLOADS, SET_UPLOAD_METADATA } from '../constants';
 import { sortByKey } from '../utils/model_utils';
 
 const initialState = {
   uploads: [],
   sortKey: null,
   view: GALLERY_VIEW,
   selectedFilters: [],
+  fetchState: false,
```

Comment:
`fetchState` is not very descriptive, especially if it's a boolean. Since this gets used elsewhere, naming it in a way that clearly communicates *which* fetch it's tracking state for would be good. Maybe `metadataFetched`?